src/tree/registries/dockerHub/DockerHubAccountTreeItem.ts

```diff
@@ -4,79 +4,76 @@
  *--------------------------------------------------------------------------------------------*/
 
 import { RequestPromiseOptions } from "request-promise-native";
-import { AzExtParentTreeItem, AzExtTreeItem, AzureWizard, GenericTreeItem, IActionContext } from "vscode-azureextensionui";
-import { dockerHubUrl, keytarConstants, PAGE_SIZE } from "../../../constants";
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "vscode-azureextensionui";
+import { dockerHubUrl, PAGE_SIZE } from "../../../constants";
 import { ext } from "../../../extensionVariables";
 import { nonNullProp } from "../../../utils/nonNull";
 import { registryRequest } from "../../../utils/registryRequestUtils";
 import { getThemedIconPath, IconPath } from "../../IconPath";
-import { isAncestoryOfRegistryType, RegistryType } from "../RegistryType";
+import { ICachedRegistryProvider, IRegistryProviderTreeItem } from "../IRegistryProvider";
+import { getRegistryContextValue, registryProviderSuffix } from "../registryContextValues";
+import { getRegistryPassword } from "../registryPasswords";
 import { DockerHubNamespaceTreeItem } from "./DockerHubNamespaceTreeItem";
-import { DockerHubPasswordStep } from "./loginWizard/DockerHubPasswordStep";
-import { DockerHubUsernameStep } from "./loginWizard/DockerHubUsernameStep";
-import { IDockerHubWizardContext } from "./loginWizard/IDockerHubWizardContext";
 
-export class DockerHubAccountTreeItem extends AzExtParentTreeItem {
-    public static contextValue: string = RegistryType.dockerHub + 'Account';
-    public contextValue: string = DockerHubAccountTreeItem.contextValue;
+export class DockerHubAccountTreeItem extends AzExtParentTreeItem implements IRegistryProviderTreeItem {
     public label: string = 'Docker Hub';
     public childTypeLabel: string = 'namespace';
     public baseUrl: string = dockerHubUrl;
-    public username?: string;
-    public password?: string;
+    public cachedProvider: ICachedRegistryProvider;
 
     private _token?: string;
-    private _nextLink: string | undefined;
-    private _isInitialized: boolean = false;
+    private _nextLink?: string;
+
+    public constructor(parent: AzExtParentTreeItem, cachedProvider: ICachedRegistryProvider) {
+        super(parent);
+        this.cachedProvider = cachedProvider;
```

Comment:
I'm not a fan of parameter properties. As discussed in [this rule](https://palantir.github.io/tslint/rules/no-parameter-properties/) (which is turned on in the functions repo):
> Parameter properties can be confusing to those new to TS as they are less explicit than other ways of declaring and initializing class members.
> It can be cleaner to keep member variable declarations in one list directly above the class constructor (instead of mixed between direct class members and constructor parameter properties).

But it's mostly a personal preference so I don't ask other people to follow that rule